sources/views/contactsData/contactsTemplate.js

```diff
@@ -74,4 +122,87 @@ export default class ContactsTemplateView extends JetView {
 			this.contactsTemplate.setValues(contact);
 		});
 	}
+
+	editContact() {
+		this.show("./contactsData.contactForm");
+	}
+
+	deleteContact() {
+		webix.confirm({
+			text: "Contact's info and all related activities will be deleted permanently! Continue?"
+		}).then(() => {
+			const contactId = this.getParam("id", true);
+			let actsToDelete;
+			let filesToDelete;
+			if (this.isNumber(contactId)) {
+				actsToDelete = activities.find(act => +act.ContactID === contactId);
+				filesToDelete = contactsFiles.find(file => +file.ContactID === contactId);
+			}
+			else if (this.isString(contactId)) {
+				actsToDelete = activities.find(act => act.ContactID.toString() === contactId);
+				filesToDelete = contactsFiles.find(file => file.ContactID.toString() === contactId);
+			}
+
+			contacts.waitSave(() => {
+				if (actsToDelete) {
+					activities.waitSave(() => {
+						actsToDelete.forEach((activity) => {
+							activities.remove(activity.id);
+						});
+					});
+				}
+				if (filesToDelete) {
+					filesToDelete.forEach((file) => {
+						contactsFiles.remove(file.id);
+					});
+				}
+				contacts.remove(contactId);
+			})
+				.then(() => {
+					webix.message("All info of the contact is deleted");
+					const firstId = contacts.getFirstId();
+					this.app.callEvent("onContactSelect", [{id: firstId}]);
+					this.show(`/top/contacts?id=${firstId}/contactsData.contactsTemplate`);
+				});
+		});
+	}
+
+	urlChange() {
+		webix.promise.all([
+			contacts.waitData,
+			activities.waitData,
+			activityTypes.waitData
+		]).then(() => {
+			const contactId = this.getParam("id", true);
+
+			if (contactId && contacts.exists(contactId)) {
+				let contactInfo = webix.copy(contacts.getItem(contactId));
+				const contactStatusId = contactInfo.StatusID;
+				const item = statuses.getItem(contactStatusId);
+				if (contactStatusId && item) {
+					contactInfo.Status = statuses.getItem(contactStatusId).Value;
+				}
+				this.contactsTemplate.setValues(contactInfo);
+
+				if (this.isNumber(contactId)) {
```

Comment:
Always compare them as strings (convert `contactId` as well) and avoid isNumber/isString (:

Reply:
fixed